weatherman.py

```diff
@@ -0,0 +1,261 @@
+def generate_barchart_for_file(file_name):
+
+    temprature_calculation_dic = { 
+        "highest_temprature_month": float("-inf"),
+        "highest_temprature_on_lowest_day": float("-inf"),
+        "lowest_temprature_month": float("inf"), 
+        "lowest_temprature_on_highest_day": float("inf")
+    }
+    file_data = open_file(file_name)
+    
+    for line_read in file_data:                         
+        temprature_calculation_dic = temprature_calculation_barchart(line_read,temprature_calculation_dic)
+
+    print_tempratures_bar(temprature_calculation_dic)
+
+
+def temprature_calculation_barchart(line_read,temprature_calculation_dict):
+    
+    if(line_read["Max TemperatureC"] != ""  and 
+         int(line_read["Max TemperatureC"]) > temprature_calculation_dict["highest_temprature_month"]  ):    
+               
+            temprature_calculation_dict["highest_temprature_month"] = int(line_read["Max TemperatureC"])
+            temprature_calculation_dict["lowest_temprature_on_highest_day"] = int(line_read["Min TemperatureC"])
+                        
+    if(line_read["Min TemperatureC"] != "" and 
+            int(line_read["Min TemperatureC"]) < temprature_calculation_dict["lowest_temprature_month"] ):        
+                                                                    
+                temprature_calculation_dict["lowest_temprature_month"] = int(line_read["Min TemperatureC"])
+                temprature_calculation_dict["highest_temprature_on_lowest_day"] = int(line_read["Max TemperatureC"])    
+    
+    return temprature_calculation_dict
+
+
+
+
+def files_traversal_year(year):
+
+        flag = False 
+        for file_name in os.listdir('.'):
+            if year in file_name:
+                generate_report_year(file_name)
+                flag = True
+
+        return flag
+
+
+def temprature_calculation_year(line_read):
+
+    highest_temprature, lowest_temprature, highest_humidity = temprature_tracker.get_max_temps()
+
+    if (line_read["Min TemperatureC"] != "" and 
+        int(line_read["Min TemperatureC"]) < lowest_temprature): 
+                                                
+        temprature_tracker.set_min_temprature(int(line_read['Min TemperatureC' ]),
+                                      line_read["PKT"])
+
+    if(line_read["Max TemperatureC"]!= ""  and 
+        int(line_read["Max TemperatureC"]) > highest_temprature):   
+        
+        temprature_tracker.set_max_temprature(int(line_read["Max TemperatureC"]),
+                                      line_read["PKT"]) 
+
+    if(line_read["Max Humidity"]!= "" and 
+        int(line_read["Max Humidity"]) > highest_humidity):
+
+        temprature_tracker.set_max_humidity(int(line_read["Max Humidity"]),
+                                          line_read["PKT"])    
+
+
+def generate_report_year(file_name):  
+
+    file_data = open_file(file_name)
+    for line_read in file_data: 
+        temprature_calculation_year(line_read)      
```

Comment:
Invoking a function takes a certain call, why can't we pass the data and get things done?

Reply:
if we move for loop inside function wouldn't it  then violate SRP i.e iterate weather data and perform temperature calculation